Sources/Filters/General/TubeFilter/test/testTubeFilter.js

```diff
@@ -181,3 +181,59 @@ test.onlyIfWebGL('Test vtkTubeFilter rendering', (t) => {
   });
   renderWindow.render();
 });
+
+test('Test vtkTubeFilter numberOfPoints', (t) => {
+  const numberOfSides = 3;
+  const numberOfLines = 2;
+  const sidesShareVertices = 1;
+
+  const polyData = vtkPolyData.newInstance();
+
+  const points = new Uint32Array([
+    0, 0, 0, 1, 1, 1, 2, 2, 2, 3, 3, 3, 4, 4, 4, 5, 5, 5,
+  ]);
+  const lines = new Uint32Array([2, 0, 1, 4, 2, 3, 4, 5]);
+
+  polyData.getPoints().setData(points);
+  polyData.getLines().setData(lines);
+
+  const tubeFilter = vtkTubeFilter.newInstance();
+  tubeFilter.setCapping(false);
+  tubeFilter.setNumberOfSides(numberOfSides);
+
+  tubeFilter.setInputData(polyData);
+
+  const tubeOutput = tubeFilter.getOutputData();
+
+  // Each segment should have numberOfPoints * sidesShareVertices * numberOfSides
+  // sidesShareVertices = 1
+  // numberOfSides = 3
+  // 2 * 1 * 3 + 4 * 1 * 3
+  t.ok(
+    tubeOutput.getPoints().getNumberOfPoints() ===
+      2 * sidesShareVertices * numberOfSides +
+        4 * sidesShareVertices * numberOfSides,
+    'Make sure the output number of points is correct.'
+  );
```

Comment:
@floryst This test should be sufficient, in the previous implementation the `npts` was incremented after calculating the number of points in the line. This would lead to the size of the first line (2) being used twice.